lua/nvim-treesitter/query_predicates.lua

```diff
@@ -87,3 +87,30 @@ end)
 
 -- Just avoid some anoying warnings for this directive
 query.add_directive('make-range!', function() end)
+
+query.add_directive('downcase!', function(match, _, bufnr, pred, metadata)
+  local text, key, value
+
+  if #pred == 3 then
+    -- (#downcase! @capture "key")
+    key = pred[3]
+    value = metadata[pred[2]][key]
+  else
+    -- (#downcase! "key")
+    key = pred[2]
+    value = metadata[key]
+  end
```

Comment:
@vigoux I made the changes that we discussed, I think I found a bug with how set was implemented in neovim. I'll be opening a PR with a fix.

Reply:
Fix is up here: https://github.com/neovim/neovim/pull/14418